roadmap/implementors-guide/src/types/overseer-protocol.md

```diff
@@ -0,0 +1,253 @@
+}
+```
+
+## Bitfield Distribution Message
+
+Messages received by the bitfield distribution subsystem.
+
+```rust
+enum BitfieldDistributionMessage {
+	/// Distribute a bitfield signed by a validator to other validators.
+	/// The bitfield distribution subsystem will assume this is indeed correctly signed.
+	DistributeBitfield(relay_parent, SignedAvailabilityBitfield),
+	/// Receive a network bridge update.
+	NetworkBridgeUpdate(NetworkBridgeEvent),
+}
+```
+
+## Bitfield Signing Message
+
+Currently, the bitfield signing subsystem receives no specific messages.
+
+```rust
+/// Non-instantiable message type
+enum BitfieldSigningMessage { }
+```
+
+## Candidate Backing Message
+
+```rust
+enum CandidateBackingMessage {
+  /// Registers a stream listener for updates to the set of backable candidates that could be backed
+  /// in a child of the given relay-parent, referenced by its hash.
+  RegisterBackingWatcher(Hash, TODO),
+  /// Note that the Candidate Backing subsystem should second the given candidate in the context of the
+  /// given relay-parent (ref. by hash). This candidate must be validated.
+  Second(Hash, CandidateReceipt),
+  /// Note a peer validator's statement about a particular candidate. Disagreements about validity must be escalated
+  /// to a broader check by Misbehavior Arbitration. Agreements are simply tallied until a quorum is reached.
+  Statement(Statement),
+}
+```
+
+## Candidate Selection Message
+
+These messages are sent from the overseer to the [Candidate Selection subsystem](../node/backing/candidate-selection.html) when new parablocks are available for validation.
+
+```rust
+enum CandidateSelectionMessage {
+  /// We recommended a particular candidate to be seconded, but it was invalid; penalize the collator.
+  Invalid(CandidateReceipt),
+}
+```
+
+## Network Bridge Message
+
+Messages received by the network bridge. This subsystem is invoked by others to manipulate access
+to the low-level networking code.
+
+```rust
+enum NetworkBridgeMessage {
+	/// Register an event producer with the network bridge. This should be done early and cannot
+	/// be de-registered.
+	RegisterEventProducer(ProtocolId, Fn(NetworkBridgeEvent) -> AllMessages),
+	/// Report a cost or benefit of a peer. Negative values are costs, positive are benefits.
+	ReportPeer(PeerId, cost_benefit: i32),
+	/// Send a message to one or more peers on the given protocol ID.
+	SendMessage([PeerId], ProtocolId, Bytes),
+}
+```
+
+## Network Bridge Update
+
+These updates are posted from the [Network Bridge Subsystem](../node/utility/network-bridge.html) to other subsystems based on registered listeners.
+
+```rust
+struct View(Vec<Hash>); // Up to `N` (5?) chain heads.
+
+enum NetworkBridgeEvent {
+	/// A peer with given ID is now connected.
+	PeerConnected(PeerId, ObservedRole), // role is one of Full, Light, OurGuardedAuthority, OurSentry
+	/// A peer with given ID is now disconnected.
+	PeerDisconnected(PeerId),
```

Comment:
does this imply guarantees that no further messages can be received from `PeerId`? Or does the consuming end need to take special care of that?

Reply:
Yes, it does. And no messages should be sent before `PeerConnected` is sent.